docs: Update io.cozy.accounts doc (#216)
cozy/cozy-doctypes: docs/io.cozy.accounts.md

    - `TWOFA_NEEDED.EMAIL`: If the two factor authentication is done by email
    - `TWOFA_NEEDED.SMS`: If the two factor authentication is done by SMS
  - `TWOFA_NEEDED_RETRY`: The 2FA code provided by the user is wrong, the user can retry by providing a new one. `TWO_FA_NEEDED_RETRY.EMAIL` and `TWO_FA_NEEDED_RETRY.SMS` can also be used.
  - `RESET_SESSION`: By finding this state, the konnector should reset the login session if there is one stored and reset the state.
- `twoFACode`: When a 2FA code is asked by the service, [Harvest](https://github.com/cozy/cozy-libs/tree/master/packages/cozy-harvest-lib) will ask the user for it from and send it to the konnector via this attribute.
- `mutedErrors`: {array} (Optional) A list of errors that that have been discarded by the user and are no longer shown in the UI. See below for more information.

## About the name of the account

We need a name to display the account to user in harvest and to create the associated folder if needed with the stack.

> Today, harvest still evaluate this `auth` attributes: `login`, `identifier`, `new_identifier` and `email` directly, if the `identifier` attribute at the root of the account is not defined.

Reviewer: ```suggestion
If missing, harvest will determine the value of the `identifier` attribute based on the following rules:
1. a konnector's manfiest field name if it has the `role: "identifier"` attribute
2. `login` if `auth.login` is defined
3. `identifier` if `auth.identifier` is defined
4. `new_identifier` if `auth.new_identifier` is defined
5. `email` if `auth.email` is defined

Once the `identifier` attribute is defined, the account name will be chosen based on these rules:
1. `auth[identifer]` if defined (we use the value of the `identifier` attibute here)
2. `_id`
```

- `twoFACode`: When a 2FA code is asked by the service, [Harvest](https://github.com/cozy/cozy-libs/tree/master/packages/cozy-harvest-lib) will ask the user for it from and send it to the konnector via this attribute.
- `mutedErrors`: {array} (Optional) A list of errors that have been discarded by the user and will no longer be shown in the UI. See below for more information.

## About the name of the account

We need a name for the account displayed to the user in harvest and for cozy-stack to create the associated folder if needed.
If missing, harvest will determine the value of the `identifier` attribute based on the following rules:
1. a konnector's manifest field name if it has the `role: "identifier"` attribute
2. `login` if `auth.login` is defined
3. `identifier` if `auth.identifier` is defined
4. `new_identifier` if `auth.new_identifier` is defined
5. `email` if `auth.email` is defined

If the `identifier` attribute is defined, the account name will be chosen based on these rules:
1. `auth[identifer]` if defined (we use the value of the `identifier` attibute here)
2. `_id`

> For banking connectors, the `identifier` attribute contains the string `identifier` so, the name of account is the `auth.identifier`.

Reviewer: ```suggestion
For banking connectors, the `identifier` attribute contains the string `identifier` so, the name of the account is the value of `auth.identifier`.
```

- `account_type`: {string} Slug of the konnector the account is related to. Note this could be a [relationship](https://github.com/cozy/cozy-doctypes/#relationships) to the konnector manifest, but we keep this field for historical reasons.
- `auth`: {object} Contains authentification data, typically a couple with `login`/`password`. It could also contain an attribute `token` for OAuth konnectors.
- `data`: {object} Additional custom data. This is useful to store arbitrary data related to the account. For instance, this can be used to store an account status, the date of the last retrieved document, etc.

> - `defaultFolderPath`: {string} Default destination folder for the konnector. Use for folder re-creation.

Reviewer 2: Used by cozy-stack for folder re-creation when the destination folder has been removed

### `auth`

> - `login` main credentials fields, displayed as default accountName. Some konnectors do not use a `login` parameter, but `identifier` or `email`. In this case the attribut role: 'identifier' should be add to parameter object in manifest.

Reviewer 3: ```suggestion
- `login` main credentials fields, displayed as default accountName. Some konnectors do not use a `login` parameter, but `identifier` or `email`. In this case the attribut role: 'identifier' should be added to parameter object in manifest.
```

Reviewer 3: Also, login is a string, right? 

Can you do like above? Something like: 
```
`login`: {string?} Main credential field.
`identifier`: {string?} If the konnector does not use `login` field. In that case, this attribute should declared as `role: identifier` in the manifest.
`email`: {string?} If the konnector does not use `login` or `identifier` fields. In that case, this attribute should declared as `role: identifier` in the manifest. 
```

Reviewer 2: /attribut/attribute/

> - `accountName` if define will be displayed preferentially than login or email or parameter with `role: 'identifier'`.

Reviewer 3: ```suggestion
- `accountName` : {string?} if defined will be displayed preferentially than login or email or parameter with `role: 'identifier'`.
```

Author: Change Done

### `auth`

> - `login`: {string} Main credentials field, displayed as default accountName. Some konnectors do not use a `login` parameter, but `identifier` or `email`. In this case the attribute role: 'identifier' should be added to parameter object in manifest.

Reviewer: Will the `identifier` be present in the account as well as the konnector manifest or can it be present in the manifest only?

Reviewer: If `identifier` or `email` can be used, we should have them appear as optional attributes here.

> - `accountName`: {string} If defined will be displayed preferentially than login or email or parameter with `role: 'identifier'`.

Reviewer: Where will it be displayed? Should we use this field to generate the konnector's folder path when `defaultFolderPath` is not defined?

- `credentials_encrypted`: {string} Store the crypted auth fields using `type: password` in manifest, typically the `password` field.

> The `auth` attributes also contain all values for the fields attributes specified in the `manifest.konnector` file.

Reviewer 3: ```suggestion
The `auth` attribute also contain all values for the fields attributes specified in the `manifest.konnector` file.
```

```json
{
  "account_type": "freemobile",
  "auth": {
    "credentials_encrypted": "aaDtwWSWsdpbbbbbbbbbbbbbbOzp...n4Y8c=",
    "login": "000000000",

>     "password": "*******" // only when setting it first time

Reviewer 4: JSON is a data-only format and doesn't support comments.
So this comment should not be inside json here (or if you choose to leave it, you can't use json type for this block code because it will fool the parser)

>     "password": "*******" // only when setting it first time

Reviewer 3: What does that mean?

Reviewer 2: I don't see the meaning of // only when setting it first time too. If this is a reference to the encryption of the password, It actually depends on the token you use to read it. I think you should remove this comment.

Reviewer 2: If credentiails_entrypted is present, password will not be present, right

Author: I remove the password fields from example.

  },

>   "defaultFolderPath": "/Administratif/Connector template/toto",

Author: Yes, I change the number in login too

    "namePath": "Free Mobile",
    "password": "*******"
  }
}
```

What we aim for:

```json
{
  "auth": {
    "credentials_encrypted": "aaDtwWSWsdpbbbbbbbbbbbbbbOzp2pBEbXlZLWjiTzOGumGRomrF2LwlRn4Y8c=",
    "login": "000000000",
    "password": "**********"
    "password": "*******" // only when setting it first time
  },
  "folderPath": "/Administratif/Free Mobile",
  "label": "freemobile",
  "namePath": "Free Mobile"
  "defaultFolderPath": "/Administratif/Connector template/toto",
  "state": "LOGIN_SUCCESS"
}

> ```

Reviewer 4: I think you should not remove end code block here

Author: Good thinker !
Thanks
